cmd/livepeer/livepeer.go

```diff
@@ -640,11 +641,16 @@ func main() {
 			if err != nil {
 				glog.Fatal("Error setting orch webhook URL ", err)
 			}
-			n.OrchestratorPool = discovery.NewWebhookPool(n, whurl)
+			n.OrchestratorPool = discovery.NewWebhookPool(whurl)
 		} else if len(orchURLs) > 0 {
-			n.OrchestratorPool = discovery.NewOrchestratorPool(n, orchURLs)
+			n.OrchestratorPool = discovery.NewOrchestratorPool(orchURLs)
 		} else if *network != "offchain" {
-			n.OrchestratorPool = discovery.NewDBOrchestratorPoolCache(n)
+			dbOrchPoolCache := discovery.NewDBOrchestratorPoolCache(dbh, n.Eth, roundsWatcher, n.Sender)
+			if originalLastSeenBlock == nil || originalLastSeenBlock.Cmp(currentRoundStartBlock) < 0 {
```

Comment:
I think making this transcoder pool caching logic post event backfill non-broadcaster specific would be helpful because it would allow an orchestrator to also rely on its DB for the state of the active set.

Reply:
That can be an option but I don't see the purpose of it as O's don't use that data for the time being? 

You could also make an argument for the opposite, saying that we should only backfill and cache the pool when the node is a B because O's currently don't have a purpose for event backfilling (perhaps for unbonding locks maybe)

Reply:
> That can be an option but I don't see the purpose of it as O's don't use that data for the time being?

Yes O doesn't use that data at the moment, but I don't think we should rule out the possibility. For example, if O had this data, then if we implement an active status check for O whenever it receives a segment (to make sure that O has access to B's reserve) then O can query its DB instead of using a contract call that involves a network request.

Furthermore, this data could generally be useful regardless of what mode that the node is running in since it could be queryable via the node's API.

Reply:
We could leverage this implementation for both O and B, but I don't think that's optimal for O. 

O would only care about `TranscoderActivated` and `TranscoderDeactivated` events pertaining to itself, not the entire set. 

Since `OrchestratorWatcher` is the writer to the DB for these events and because we already run `OrchestratorWatcher` when `NodeType == OrchestratorNode`, I suggest we do the following:
- Add an address filter option to `OrchestratorWatcher` for events only belonging to certain orchestrators
- Add an `isActive() (bool, error)` method to `core.Orchestrator` that queries `db.SelectOrchs` with a `currentRound` filter., since O will only write events about itself to the database this query will return 1 result when O is active and nil when O is inactive
- Leverage `orch.isActive()` in `server/ServeSegment()`

Reply:
> O would only care about TranscoderActivated and TranscoderDeactivated events pertaining to itself, not the entire set.

Yep good point.

I do think it could be a useful for a node, regardless of whether it is running in broadcaster mode or orchestrator mode, to be able to compute the active set for the current round. For example, a developer building on top of the node's API hitting an endpoint that responds with the active set. What do you think?

Reply:
I don't think we have put in sufficient work and thought into this to make a decision on this right now. 

We still have to throughly think about what we want such API to look like and perhaps even what APIs we want developers to use for what data. 

To illustrate: would it be worth the hassle for an orchestrator to periodically query other orchestrators for their info, just to have that off-chain orchestrator data available for an API endpoint or would an orchestrator list view fetched through eg. the livepeer subgraph suffice ? One should also consider how much this endpoint will actually be hit compared to the continuous computational resources it expends as there are alternatives for it.

Reply:
I agree with those points.

> would it be worth the hassle for an orchestrator to periodically query other orchestrators for their info, just to have that off-chain orchestrator data available for an API endpoint

To clarify I was *not* suggesting also having the orchestrator fetch off-chain info of other orchestrators. I was only suggesting having the orchestrator compute the state of the active set i.e. the addresses in the set.

> One should also consider how much this endpoint will actually be hit compared to the continuous computational resources it expends as there are alternatives for it.

To clarify I was *not* suggesting we focus on creating that endpoint now. I was only mentioning one potential use case for the node to always be aware of the state of the current active set. 

I hear you on not trying to predict the usefulness of a node always being aware of the current active set - if it ends up being necessary we can make the change when the time comes and it shouldn't be that involved of a change.

We can use your suggestion of adding a filter to `OrchestratorWatcher` when the node is in orchestrator mode, but I recommend doing that outside of this PR.